tests/parser/test_jsonschema.py

```diff
@@ -914,6 +914,28 @@ def test_has_ref_with_schema_keywords_extras_with_metadata_only_keys() -> None:
     assert obj.has_ref_with_schema_keywords is False
 
 
+def test_has_ref_with_schema_keywords_extras_with_extension_keys() -> None:
+    """Test has_ref_with_schema_keywords when extras contains only x-* extension keys.
+
+    OpenAPI/JSON Schema extension fields (x-*) should be treated as metadata
+    and not trigger schema merging, which prevents infinite recursion with
```

Comment:
> infinite recursion

Is recursion for #2800 infinite, or just high enough to hit recursion limits?

Reply:
It is not truly infinite. It recurses many times and finally hits Python recursion limit, so it looks like infinite. This  change avoids that path.